node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -283,37 +286,56 @@ impl Initialized {
 		self.participation.process_active_leaves_update(ctx, &update).await?;
 
 		if let Some(new_leaf) = update.activated {
-			match self
-				.rolling_session_window
-				.cache_session_info_for_head(ctx.sender(), new_leaf.hash)
-				.await
-			{
-				Err(e) => {
-					gum::warn!(
+			let session_idx =
+				self.runtime_info.get_session_index_for_child(ctx.sender(), new_leaf.hash).await;
+
+			match session_idx {
+				Ok(session_idx)
+					if self.gaps_in_cache || session_idx > self.highest_session_seen =>
+				{
+					// If error has occurred during last session caching - fetch the whole window
+					// Otherwise - cache only the new sessions
+					let lower_bound = if self.gaps_in_cache {
+						session_idx.saturating_sub(DISPUTE_WINDOW.get() - 1)
+					} else {
+						self.highest_session_seen
```

Comment:
nit: should it be `self.highest_session_seen + 1`? I guess doesn't matter much.

Reply:
Fixed